### ISSUE TITLE
Fix FP S3498: should not raise on Aura Lightning Components

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#2885]
The rule currently raises issues on Salesforce Aura Lightning Components (see [Peach](https://peach.sonarsource.com/project/issues?id=sf-LightningFlowComponents_nodev10&open=AX0khViHTPFHcz6tHEez&resolved=false&rules=javascript%3AS3498)) like the following:

```javascript
({
  action: function(component, event, helper) {
    /* .... */
  }
});
```

Although such issues are true positives, all snippets from [Salesforce documentation](https://developer.salesforce.com/docs/atlas.en-us.lightning.meta/lightning/expr_resource_value_provider.htm) use this old syntax as well as code generators, and make the rule noisy for Aura Lightning Components. Therefore, we should add an exception to the rule for these components.
